crates/c-api/src/linker.rs

```diff
@@ -120,3 +120,31 @@ pub unsafe extern "C" fn wasmtime_linker_get_default(
         *func = Box::into_raw(Box::new(HostRef::new(linker.store(), f).into()))
     })
 }
+
+#[no_mangle]
+pub unsafe extern "C" fn wasmtime_linker_get_one_by_name(
```

Comment:
I think the `unsafe` here can be dropped?

Reply:
Good point! I took the liberty of adding a new commit cleaning up a few other unnecessary `unsafe` qualifiers around the C API as well.